src/agentscope/models/deepseek_model.py

```diff
@@ -0,0 +1,320 @@
+from ..message import Msg
+import time
+from ._model_utils import (
+    _verify_reasoning_content_in_openai_delta_response,
+    _verify_reasoning_content_in_openai_message_response,
+    _verify_text_content_in_openai_delta_response,
+    _verify_text_content_in_openai_message_response,
+)
+
+
+class DeepSeekWrapperBase(ModelWrapperBase, ABC):
+    """The model wrapper for DeepSeek API."""
+
+    def __init__(
+        self,
+        config_name: str,
+        model_name: Optional[str] = None,
+        api_key: Optional[str] = None,
+        client_args: Optional[dict] = None,
+        generate_args: Optional[dict] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initalize the openai client.
+        Args:
+            config_name (`str`):
+                The name of the model configuration.
+            model_name (`Optional[str]`):
+                The name of the model. If not provided, the default model
+                will be used.
+            api_key (`Optional[str]`):
+                The API key for the openai client.
+            client_args (`Optional[dict]`):
+                The arguments for the openai client.
+            generate_args (`Optional[dict]`):
+                The arguments for the generation. e.g. `temperature`,
+                `max_tokens`
+        """
+
+        super().__init__(
+            config_name=config_name,
+            model_name=model_name,
+        )
+
+        self.generate_args = generate_args or {}
+
+        try:
+            import openai
+        except ImportError as e:
+            raise ImportError(
+                "Cannot find openai package, please install it by"
+                "`pip install openai`",
+            ) from e
+
+        DEEPSEEK_URL_API: str = "https://api.deepseek.com"
+        # If deepseek official api is not available, use aliyun hosted deepseek
+        # remember to change the model name to deepseek-r1 and deepseek-v3
+        # DEEPSEEK_URL_API: str = (
+        #     "https://dashscope.aliyuncs.com/compatible-mode/v1"
+        # )
+        # Fixed max length for DeepSeek API
+        DEEPSEEK_MAX_LENGTH: int = 8192
+
+        self.client = openai.OpenAI(
+            api_key=api_key, base_url=DEEPSEEK_URL_API, **(client_args or {})
+        )
+        self.max_length = DEEPSEEK_MAX_LENGTH
+
+    def format(
+        self,
+        *args: Union[Msg, Sequence[Msg]],
+    ) -> Union[List[dict], str]:
+        raise RuntimeError(
+            f"Model Wrapper [{type(self).__name__}] doesn't "
+            f"need to format the input. Please try to use the "
+            f"model wrapper directly.",
+        )
+
+
+class DeepSeekChatWrapper(DeepSeekWrapperBase):
+    """The model wrapper for DeepSeek Chat API."""
+
+    model_type: str = "deepseek_chat"
+    # Recommend use aliyun hosted deepseek
+    # then the model name is deepseek-r1 and deepseek-v3
+
+    # supported_models: list[str] = [
+    #     "deepseek-chat",
```

Comment:
Since we don't hope developers to modify this file, it's suggested to add `deepseek-r1` and `deepseek-v3` into the supported models.